stairway/src/main/java/bio/terra/stairway/impl/FlightFilterAccess.java

```diff
@@ -289,19 +278,66 @@ private void storeFlightPredicateValue(
    */
   @VisibleForTesting
   String makeInputPredicateSql(FlightFilterPredicate predicate) {
-    return "(I.key = '"
+    String valuePredicate = "";
+    if (predicate.getDatatype() == Datatype.NULL) {
+      return "(EXISTS (SELECT 0 FROM flightinput I WHERE F.flightid = I.flightid AND I.key = "
```

Comment:
If the subselect is the same, is there a way to refactor the string to avoid the code duplication?

Reply:
It's not totally the same and pulling out the similar pieces I thought would make the code less readable